### PR TITLE
Cont. Systems Details Page Live Data Implementation 

### LINKED ISSUE #94: Function details - Automorphism and rational pre-periodic table details
**Is this task related to a feature or a bug? Please include relevant issue numbers below.**
This task contributes to resolving issue #29

**Describe the solution you'd like**
 Automorphism and rational pre-periodic table data the details page is now static and with the new database, we need to get the actual citation information.

image

- Make it part of a single API call that would get all the information related to the details page.
- We do not want to make multiple API calls for a single details page.

### PR DESCRIPTION
**Pull request by Evan-Richmond**
Fixes #94

What was changed?

The automorphism group table, critical points table, info table, Models table, and the rational points table were restructured to use live data passed by the systemdetails.js file which gets the data from the postgres connector using the getsystem method. These tables are passed a data dictionary object to reference the different columns of the relevant database tables within the tables. Only one API call is made, and the data obtained can be passed to multiple tables for population.

Backend files were updated to follow linting standards for cleanliness of code and readability for future editors.

Why was it changed?

- The previous tables only contained static information and contained no information about the actual system

- Files were edited for readability and cleanliness to accord with linting standards

How was it changed?

Updated the structure of the table to accept a data object, passed this data in systemdetails, and referenced the data object within the table itself to populate the table.

The Models table inparticular relies on mapping the appropriate tokenizations of the values of each _model entry  in the dictionary to its appropriate column per model row.

Since the additions of the hutz_dev branch which this branch is based off of, I also restructured the getSystem method to return a dictionary and the newly formatted label as desired 

Screenshots that show the changes (if applicable):

image

image

image

Note many temp values to be updated in final PR, need to check with Dr. Hutz about formatting and what to display. 

image

Note this sprint all 3 team members have collaboratively worked in the same branch to perform linting fixes alongside the details page live data implementation.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -170,10 +172,10 @@ include-naming-hint=no
 property-classes=abc.abstractproperty,cached_property.cached_property,cached_property.threaded_cached_property,cached_property.cached_property_with_ttl,cached_property.threaded_cached_property_with_ttl
 
 # Regular expression matching correct function names
-function-rgx=^(?P<snake_case>_?[a-z][a-z0-9_]*)$
+function-rgx=^(?P<snake_case>_?[a-z][a-zA-Z0-9_]*)$
 
 # Regular expression matching correct variable names
-variable-rgx=^[a-z][a-z0-9_]*$
+variable-rgx=^[a-z][a-zA-Z0-9_]*$
```
Comment (AbhilashKotha) [inline .pylintrc:178]: Same here, change the variable names, not the rule.

File: .pylintrc [hunk 2]
```diff
@@ -182,7 +184,7 @@ const-rgx=^(_?[A-Z][A-Z0-9_]*|__[a-z0-9_]+__|_?[a-z][a-z0-9_]*)$
 attr-rgx=^_{0,2}[a-z][a-z0-9_]*$
 
 # Regular expression matching correct argument names
-argument-rgx=^[a-z][a-z0-9_]*$
+argument-rgx=^[a-z][a-zA-Z0-9_]*$
```
Comment (AbhilashKotha) [inline .pylintrc:187]: Same here, change the argument names, not the rule.

File: .pylintrc [hunk 3]
```diff
@@ -194,10 +196,10 @@ inlinevar-rgx=^[a-z][a-z0-9_]*$
 class-rgx=^(?P<camel_case>_{0,2}[A-Z][a-zA-Z0-9_]*)$
 
 # Regular expression matching correct module names
-module-rgx=^(_?[a-z][a-z0-9_]*|__init__)$
+module-rgx=^(_?[a-zA-Z][a-zA-Z0-9_]*|__init__)$
```
Comment (AbhilashKotha) [inline .pylintrc:199]: same here

File: .pylintrc [hunk 4]
```diff
@@ -194,10 +196,10 @@ inlinevar-rgx=^[a-z][a-z0-9_]*$
...
 
 # Regular expression matching correct method names
-method-rgx=(?x)^(?:(?P<exempt>_[a-z0-9_]+__|runTest|setUp|tearDown|setUpTestCase|tearDownTestCase|setupSelf|tearDownClass|setUpClass|(test|assert)_*[A-Z0-9][a-zA-Z0-9_]*|next)|(?P<camel_case>_{0,2}[A-Z][a-zA-Z0-9_]*)|(?P<snake_case>_{0,2}[a-z][a-z0-9_]*))$
+method-rgx=(?x)^(?:(?P<exempt>_[a-z0-9_]+__|runTest|setUp|tearDown|setUpTestCase|tearDownTestCase|setupSelf|tearDownClass|setUpClass|(test|assert)_*[A-Z0-9][a-zA-Z0-9_]*|next)|(?P<camel_case>_{0,2}[A-Z][a-zA-Z0-9_]*)|(?P<snake_case>_{0,2}[a-z][a-zA-Z0-9_]*))$
```
Comment (AbhilashKotha) [inline .pylintrc:202]: same here

File: Backend/PostgresConnector.py [hunk 5]
```diff
@@ -79,15 +129,15 @@ def getFilteredSystems(self,filters):
...
+                                    elif i == 1:
                                         mon.append('x^'+str(d-i) + 'y')
                                     elif (d-i) == 1:
                                         mon.append('x' + 'y^' + str(i))
                                     else:
                                         mon.append('x^'+str(d-i) + 'y^'+str(i))
                             mon_dict[d] = mon
-                        poly = '['
-                        c = row[2]
+                        poly = []
```
Comment (AbhilashKotha) [inline Backend/PostgresConnector.py:139]:
```suggestion
                        poly = '['
```

**Review by AbhilashKotha — CHANGES_REQUESTED** [review]
Please address my comments regarding the linking and other things.

**Review by lRolwes — APPROVED** [review]

**Review by AbhilashKotha — APPROVED** [review]

**Pull request merged**